src/main/java/org/terasology/segmentedpaths/events/OnExitSegment.java

```diff
@@ -18,14 +18,27 @@
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.Event;
 
+/**
+ * Event called when an entity which has {@link org.terasology.segmentedpaths.components.PathFollowerComponent} leaves a segment.
+ */
 public class OnExitSegment implements Event {
 
+    /**
+     * EntityRef to an entity, which leaves a segment.
+     */
     private EntityRef pathFollowingEntity;
 
+    /**
+     * Event constructor.
+     * @param pathFollowingEntity which leaves a segment.
+     */
     public OnExitSegment(EntityRef pathFollowingEntity){
         this.pathFollowingEntity = pathFollowingEntity;
     }
 
+    /**
+     * @return the entity, which leaves a segment.
```

Comment:
Where is the description of the method?
`Returns the entity ...` or something. I realize that it seems kinda redundant, but every javadoc should have a short description of the method, then explain all params, then explain return value, then (if applicable) state all exceptions.